spec/lib/jobs/dell-wsman-inventory-spec.js

```diff
@@ -0,0 +1,269 @@
+    var WsmanInventoryJobClass;
+    var wsmanInventoryJob;
+    var wsmanBaseJob;
+    var wsmanTool;
+    var logger;
+    var configuration;
+    var mockWaterline;
+    var loggerSpy;
+    var doneSpy;
+
+    var obm = {
+        id: '59c4797e70bf1d7c172930dc',
+        node: '/api/2.0/nodes/59c4720870bf1d7c172930db',
+        service: 'dell-wsman-obm-service',
+        config: {
+            user: "admin",
+            host: "172.31.128.73"
+        }
+    };
+
+     var dellConfigs = {
+         "wsmanCallbackUri": "http://172.31.128.1:9988/api/2.0/wsmanCallback/_IDENTIFIER_",
+         "gateway": "http://localhost:46011","services": {
+             "inventory": {
+                "serverCallback": "http://localhost:46011/api/1.0/server/inventory/callback",
+            }
+        }
+    };
+
+    before(function() {
+        helper.setupInjector([
+            helper.require('/lib/utils/job-utils/wsman-tool.js'),
+            helper.require('/lib/jobs/base-job.js'),
+            helper.require('/lib/jobs/dell-wsman-base-job'),
+            helper.require('/lib/jobs/dell-wsman-inventory.js'),
+        ]);
+        WsmanInventoryJobClass = helper.injector.get('Job.Dell.Wsman.Inventory');
+        configuration = helper.injector.get('Services.Configuration');
+        wsmanTool = helper.injector.get('JobUtils.WsmanTool');
+        mockWaterline = helper.injector.get('Services.Waterline');
+        wsmanBaseJob = helper.injector.get('Job.Dell.Wsman.Base');
+        wsmanInventoryJob = new WsmanInventoryJobClass({}, {}, uuid.v4());
+        logger = helper.injector.get('Logger');
+        loggerSpy = sinon.spy(logger.prototype, 'error');
+        doneSpy = sinon.spy(wsmanInventoryJob, '_done');
+        this.sandbox = sinon.sandbox.create();
+    });
+
+    beforeEach(function() {
+        this.sandbox.stub(wsmanBaseJob.prototype, 'checkOBM');
+        this.sandbox.stub(wsmanBaseJob.prototype, 'getIpAddress');
+        this.sandbox.stub(wsmanBaseJob.prototype, '_subscribeHttpResponseUuid');
+        this.sandbox.stub(configuration, 'get');
+        this.sandbox.stub(WsmanInventoryJobClass.prototype, '_handleAsyncRequest');
+        this.sandbox.stub(WsmanInventoryJobClass.prototype, 'handleAsyncResponse');
+        mockWaterline.catalogs = {
+            findLatestCatalogOfSource: this.sandbox.stub(),
+            updateByIdentifier: this.sandbox.stub(),
+            create: this.sandbox.stub()
+        };
+        this.sandbox.stub(wsmanTool.prototype, 'clientRequest');
+    });
+
+    afterEach(function() {
+        this.sandbox.restore();
+    });
+
+    describe('initJob', function() {
+        it('should handle the case of lacking dell configuration.', function() {
+            wsmanBaseJob.prototype.checkOBM.resolves(obm);
+            configuration.get.withArgs('dell').returns(undefined);
+            expect(wsmanInventoryJob._initJob()).to.be.rejectedWith('Dell Inventory web service is not defined in wsmanConfig.json.');
```

Comment:
If there is no SMI service uri defined in smiConifg.json. What will happen?

Reply:
Good suggestion. We could add some more tests to cover such cases later.